pkg/server/datastore/sqlstore/sqlstore.go

```diff
@@ -3874,3 +3883,9 @@ func lookupSimilarEntry(ctx context.Context, db *sqlDB, tx *gorm.DB, entry *comm
 		return nil, nil
 	}
 }
+
+// roundCreatedAtInSeconds rounds the createdAt time to the nearest second, and return the time in seconds since the
+// unix epoch. This function is used to avoid issues with databases versions that do not support sub-second precision.
+func roundedCreatedAtInSeconds(createdAt time.Time) int64 {
```

Comment:
sqlstore tests with mysql version 5.7.1 were failing (it does not support sub-second precision), and that's because the createdAt value returned by create method was different from the one returned by fetch method in one second. Going a little deeper into the code I discovered that gorm rounds the time before saving it in the database (using half_up), so when the value is fetched, the time retrieved is already rounded. The values from create and fetch turns out to be different because`.Unix()` method doesn't round the time if it has the sub-second precision. 

My solution here doesn't round the value before writing it to the database, only round the value in the model-entry mapper functions, so the real data remains with the precision supported by the database.

![image](https://user-images.githubusercontent.com/33766735/225440637-97fcb73c-00b5-4643-bbd9-581cc96bff36.png)
![image](https://user-images.githubusercontent.com/33766735/225440733-e69ae284-eae1-4032-b712-5ce168ec3f70.png)
![image](https://user-images.githubusercontent.com/33766735/225440850-91d5872a-37bb-485d-8832-b40b817ed389.png)

Reply:
I see, so in this case we are not losing precision but rather making sure that the rounding matches what gorm does under the covers. Thanks for explaining.